File: local-data-persistent/LatihanRoom/java/app/src/main/java/com/dicoding/picodiploma/mynoteapps/ui/insert/NoteAddUpdateViewModel.java
package com.dicoding.picodiploma.mynoteapps.ui.insert;

import android.app.Application;
import androidx.lifecycle.ViewModel;

import com.dicoding.picodiploma.mynoteapps.database.Note;
import com.dicoding.picodiploma.mynoteapps.repository.NoteRepository;

public class NoteAddUpdateViewModel extends ViewModel {

    private final NoteRepository mNoteRepository;

    public NoteAddUpdateViewModel(Application application) {
        mNoteRepository = new NoteRepository(application);
    }

    public void insert(Note note) {
        mNoteRepository.insert(note);
    }

    public void update(Note note) {
        mNoteRepository.update(note);
    }

    public void delete(Note note) {
        mNoteRepository.delete(note);
    }
}

File: local-data-persistent/LatihanRoom/kotlin/app/src/main/java/com/dicoding/picodiploma/mynoteapps/ui/insert/NoteAddUpdateViewModel.kt
package com.dicoding.picodiploma.mynoteapps.ui.insert

import android.app.Application
import androidx.lifecycle.ViewModel
import com.dicoding.picodiploma.mynoteapps.database.Note
import com.dicoding.picodiploma.mynoteapps.repository.NoteRepository

class NoteAddUpdateViewModel(application: Application) : ViewModel() {

    private val mNoteRepository: NoteRepository = NoteRepository(application)

    fun insert(note: Note) {
        mNoteRepository.insert(note)
    }

    fun update(note: Note) {
        mNoteRepository.update(note)
    }

    fun delete(note: Note) {
        mNoteRepository.delete(note)
    }

}